### PR TITLE
refactor of watchers to use map functions

### PR DESCRIPTION
**Pull request by ingmargoudt**

### CODE REVIEW

File: Mage/src/main/java/mage/watchers/common/CastSpellLastTurnWatcher.java [hunk 1]
```diff
@@ -98,20 +90,11 @@ public void reset() {
     }
 
     public int getAmountOfSpellsAllPlayersCastOnCurrentTurn() {
-        int totalAmount = 0;
-        for (Integer amount : amountOfSpellsCastOnCurrentTurn.values()) {
-            totalAmount += amount;
-        }
-        return totalAmount;
+        return amountOfSpellsCastOnCurrentTurn.values().stream().mapToInt(Integer::intValue).sum();
```
Comment (NorthFury) [inline Mage/src/main/java/mage/watchers/common/CastSpellLastTurnWatcher.java]: `amountOfSpellsCastOnCurrentTurn.values().stream().reduce(0, Integer::sum);`

It's not much different but it's how I would do it.

File: Mage/src/main/java/mage/watchers/common/CommanderInfoWatcher.java [hunk 2]
```diff
@@ -77,10 +77,7 @@ public void watch(GameEvent event, Game game) {
                 DamagedPlayerEvent damageEvent = (DamagedPlayerEvent) event;
                 if (damageEvent.isCombatDamage()) {
                     UUID playerUUID = event.getTargetId();
-                    Integer damage = damageToPlayer.get(playerUUID);
-                    if (damage == null) {
-                        damage = 0;
-                    }
+                    Integer damage = damageToPlayer.getOrDefault(playerUUID,0);
```
Comment (NorthFury) [inline Mage/src/main/java/mage/watchers/common/CommanderInfoWatcher.java]: formatting

File: Mage/src/main/java/mage/watchers/common/MiracleWatcher.java [hunk 3]
```diff
@@ -74,14 +75,9 @@ public void watch(GameEvent event, Game game) {
...
-                if (amount == null) {
-                    amount = 1;
-                } else {
-                    amount++;
-                }
-                amountOfCardsDrawnThisTurn.put(playerId, amount);
-                if (amount == 1) {
+                amountOfCardsDrawnThisTurn.putIfAbsent(playerId, 0);
+                amountOfCardsDrawnThisTurn.compute(playerId, (p, amount) -> amount + 1);
+                if (amountOfCardsDrawnThisTurn.get(playerId) == 1) {
```
Comment (NorthFury) [inline Mage/src/main/java/mage/watchers/common/MiracleWatcher.java]: `Integer amount = 1 + amountOfCardsDrawnThisTurn.getOrDefault(playerId, 0);
amountOfCardsDrawnThisTurn.put(playerId, amount);

if (amount == 1) {
  // do stuff
}`

File: Mage/src/main/java/mage/watchers/common/PlayerAttackedWatcher.java [hunk 4]
```diff
@@ -66,8 +66,8 @@ public PlayerAttackedWatcher copy() {
     @Override
     public void watch(GameEvent event, Game game) {
         if (event.getType() == GameEvent.EventType.ATTACKER_DECLARED) {
-            int numberAttackers = playerAttacked.containsKey(event.getPlayerId()) ? playerAttacked.get(event.getPlayerId()) : 0;
-            playerAttacked.put(event.getPlayerId(), numberAttackers + 1);
+            playerAttacked.putIfAbsent(event.getPlayerId(), 0);
+            playerAttacked.compute(event.getPlayerId(), (p,amount)->amount + 1);
```
Comment (NorthFury) [inline Mage/src/main/java/mage/watchers/common/PlayerAttackedWatcher.java]: formatting

File: Mage/src/main/java/mage/watchers/common/PlayerGainedLifeWatcher.java [hunk 5]
```diff
@@ -65,23 +65,14 @@ public void watch(GameEvent event, Game game) {
...
             if (playerId != null) {
-                Integer amount = amountOfLifeGainedThisTurn.get(playerId);
-                if (amount == null) {
-                    amount = event.getAmount();
-                } else {
-                    amount = amount + event.getAmount();
-                }
-                amountOfLifeGainedThisTurn.put(playerId, amount);
+                amountOfLifeGainedThisTurn.putIfAbsent(playerId, 0);
+                amountOfLifeGainedThisTurn.compute(playerId, (p, amount) -> amount +event.getAmount());
```
Comment (NorthFury) [inline Mage/src/main/java/mage/watchers/common/PlayerGainedLifeWatcher.java]: formatting

File: Mage/src/main/java/mage/watchers/common/CardsAmountDrawnThisTurnWatcher.java [hunk 6]
```diff
@@ -61,11 +60,8 @@ public CardsAmountDrawnThisTurnWatcher(final CardsAmountDrawnThisTurnWatcher wat
     @Override
     public void watch(GameEvent event, Game game) {
         if (event.getType() == GameEvent.EventType.DREW_CARD) {
-            if (amountOfCardsDrawnThisTurn.containsKey(event.getPlayerId())) {
-                amountOfCardsDrawnThisTurn.put(event.getPlayerId(), amountOfCardsDrawnThisTurn.get(event.getPlayerId()) + 1);
-            } else {
-                amountOfCardsDrawnThisTurn.put(event.getPlayerId(), 1);
-            }
+            amountOfCardsDrawnThisTurn.putIfAbsent(event.getPlayerId(), 0);
+            amountOfCardsDrawnThisTurn.compute(event.getPlayerId(), (k, amount) -> amount += 1);
```
Comment (ingmargoudt) [inline Mage/src/main/java/mage/watchers/common/CardsAmountDrawnThisTurnWatcher.java]: agree, will update that

File: Mage/src/main/java/mage/watchers/common/AmountOfDamageAPlayerReceivedThisTurnWatcher.java [hunk 7]
```diff
@@ -31,17 +31,16 @@
...
 import mage.game.Game;
 import mage.game.events.GameEvent;
 import mage.watchers.Watcher;
 
 /**
- *
  * @author jeffwadsworth
- *
- * Amount of damage received by a player this turn
- *
```
Comment (ingmargoudt) [inline Mage/src/main/java/mage/watchers/common/AmountOfDamageAPlayerReceivedThisTurnWatcher.java]: I blame my IntelliJ for that. Will revert it.

File: Mage/src/main/java/mage/watchers/common/CardsPutIntoGraveyardWatcher.java [hunk 8]
```diff
@@ -74,13 +74,9 @@ public void watch(GameEvent event, Game game) {
...
             if (playerId != null && game.getCard(event.getTargetId()) != null) {
-                Integer amount = amountOfCardsThisTurn.get(playerId);
-                if (amount == null) {
-                    amount = 1;
-                } else {
-                    ++amount;
-                }
-                amountOfCardsThisTurn.put(playerId, amount);
+                amountOfCardsThisTurn.putIfAbsent(playerId, 0);
+                amountOfCardsThisTurn.compute(playerId, (k, amount) -> amount += 1);
```
Comment (ingmargoudt) [inline Mage/src/main/java/mage/watchers/common/CardsPutIntoGraveyardWatcher.java]: agree, will update that

File: Mage/src/main/java/mage/watchers/common/MiracleWatcher.java (hunk 3 above)
Comment (ingmargoudt) [inline Mage/src/main/java/mage/watchers/common/MiracleWatcher.java]: Nice! Will update that

**Comment by NorthFury** [comment]
The more I think about it, the more I like the variant using `getOrDefault` and `put` versus, `putIfAbsent and compute`

File: Mage/src/main/java/mage/watchers/common/FirstSpellCastThisTurnWatcher.java [hunk 9]
```diff
@@ -58,10 +58,6 @@ public void reset() {
     }
 
     public UUID getIdOfFirstCastSpell(UUID playerId) {
-        if (playerFirstSpellCast.get(playerId) == null) {
-            return playerFirstCastSpell.get(playerId);
-        } else {
-            return playerFirstSpellCast.get(playerId);
-        }
+        return playerFirstSpellCast.getOrDefault(playerId, playerFirstSpellCast.get(playerId));
```
Comment (kubikrubikvkube) [inline Mage/src/main/java/mage/watchers/common/FirstSpellCastThisTurnWatcher.java]: This code block was pretty weird. We are trying to get value from map, and if map does not contain value it still tryies to get value?
Maybe 
`return playerFirstSpellCast.getOrDefault(playerId, null);`
is the correct option?

**Comment by kubikrubikvkube** [comment]
Everything else looks good for me

**Comment by ingmargoudt** [comment]
Ran all tests, no regression encountered

**Pull request merged**